ocaml-lsp-server/src/inference.ml

```diff
@@ -0,0 +1,68 @@
+open Import
+
+let infer_intf_for_impl doc =
+  match Document.kind doc with
+  | Intf ->
+    Fiber.return
+      (Error
+         (Invalid_argument "the provided document is not an implementation."))
```

Comment:
We don't use `Invalid_argument` in our code base. If it's a programming error, then it should be `Code_error.E`, if it's something for the user, it should be `Jsonrpc.Response.Error.t`